src/fabric-1.20.1/templates/elementinits/biomes.java.ftl

```diff
@@ -61,20 +64,129 @@ public class ${JavaModName}Biomes {
+							Climate.Parameter.span(${biome.genErosion.min}f, ${biome.genErosion.max}f),
+							Climate.Parameter.span(0.2f, 0.9f),
+							Climate.Parameter.span(${biome.genWeirdness.min}f, ${biome.genWeirdness.max}f),
+							0 <#-- offset -->
+						),
+						biomeRegistry.getHolderOrThrow(ResourceKey.create(Registries.BIOME, new ResourceLocation("${modid}", "${biome.getModElement().getRegistryName()}")))
+					));
+					</#list>
+
+					chunkGenerator.biomeSource = MultiNoiseBiomeSource.createFromList(new Climate.ParameterList<>(parameters));
+					chunkGenerator.featuresPerStep = Suppliers.memoize(() ->
+							FeatureSorter.buildFeaturesPerStep(List.copyOf(chunkGenerator.biomeSource.possibleBiomes()), biome ->
+									chunkGenerator.generationSettingsGetter.apply(biome).features(), true));
+				}
+				
+				// Inject surface rules
+				if(chunkGenerator instanceof NoiseBasedChunkGenerator noiseGenerator) {
+					NoiseGeneratorSettings noiseGeneratorSettings = noiseGenerator.settings.value();
+					((NoiseGeneratorSettingsAccess)(Object)noiseGeneratorSettings).addSurfaceRules(SurfaceRules.sequence(
+						${JavaModName}SurfaceRules.makeOverworldRules(), noiseGeneratorSettings.surfaceRule()));
+				}
+			}
 			</#if>
-		</#list>
-
-		<#if spawn_overworld?has_content>
-			SurfaceRuleManager.addSurfaceRules(SurfaceRuleManager.RuleCategory.OVERWORLD, ${JavaModName}.MODID, ${JavaModName}SurfaceRules.makeOverworldRules());
-		</#if>
-
-		<#if spawn_nether?has_content>
-			SurfaceRuleManager.addSurfaceRules(SurfaceRuleManager.RuleCategory.NETHER, ${JavaModName}.MODID, ${JavaModName}SurfaceRules.makeNetherRules());
-		</#if>
+			
+			<#if spawn_nether?has_content>
+			if(dimensionType == dimensionTypeRegistry.getOrThrow(BuiltinDimensionTypes.NETHER)) {
+				ChunkGenerator chunkGenerator = levelStem.generator();
+
+				// Inject biomes to biome source
+				if(chunkGenerator.getBiomeSource() instanceof MultiNoiseBiomeSource noiseSource) {
+					List<Pair<Climate.ParameterPoint, Holder<Biome>>> parameters = new ArrayList<>(noiseSource.parameters().values());
+
+					<#list spawn_nether as biome>
+					parameters.add(new Pair<>(
+						new Climate.ParameterPoint(
+							Climate.Parameter.span(${biome.genTemperature.min}f, ${biome.genTemperature.max}f),
+							Climate.Parameter.span(${biome.genHumidity.min}f, ${biome.genHumidity.max}f),
+							Climate.Parameter.span(${biome.genContinentalness.min}f, ${biome.genContinentalness.max}f),
+							Climate.Parameter.span(${biome.genErosion.min}f, ${biome.genErosion.max}f),
+							Climate.Parameter.point(0.0f),
+							Climate.Parameter.span(${biome.genWeirdness.min}f, ${biome.genWeirdness.max}f),
+							0 <#-- offset -->
+						),
+						biomeRegistry.getHolderOrThrow(ResourceKey.create(Registries.BIOME, new ResourceLocation("${modid}", "${biome.getModElement().getRegistryName()}")))
+					));
+					parameters.add(new Pair<>(
+						new Climate.ParameterPoint(
```

Comment:
Instead of duplicating this code every time, maybe you should make a variable and use it every time where it's needed.

Reply:
But this is how it's done in the core generators, I copied the code from there

Reply:
see https://github.com/MCreator/MCreator/blob/master/plugins/generator-1.20.1/forge-1.20.1/templates/elementinits/biomes.java.ftl

Reply:
Oh ok, find then.